bob-common/src/name_types.rs

```diff
@@ -0,0 +1,235 @@
+use std::{
+    fmt::{Debug, Display, Formatter, Result as FmtResult},
+    hash::{Hash, Hasher},
+    sync::Arc,
+};
+use serde::{
+    de::{Visitor, Deserialize, Deserializer},
+    ser::{Serialize, Serializer}
+};
+
+/// Node name struct. Clone is lightweight
+#[derive(Clone)]
+pub struct NodeName(Arc<str>);
+
+/// Disk name struct. Clone is lightweight
+#[derive(Clone)]
+pub struct DiskName(Arc<str>);
+
+
+macro_rules! impl_str_partial_eq {
+    ($ty:ty, $other:ty) => {
+        impl<'a> PartialEq<$other> for $ty {
+            fn eq(&self, other: &$other) -> bool {
+                PartialEq::eq(AsRef::<str>::as_ref(self), AsRef::<str>::as_ref(other))
+            }
+        }
+    };
+}
+
+/// Visistor for deserialization
+struct ArcStrVisitor;
+
+impl<'de> Visitor<'de> for ArcStrVisitor {
+    type Value = Arc<str>;
+
+    fn expecting(&self, formatter: &mut Formatter) -> FmtResult {
+        formatter.write_str("string")
+    }
+    
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error, 
+    {
+        Ok(v.into())
+    }
+
+    fn visit_borrowed_str<E>(self, v: &'de str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error, 
+    {
+        Ok(v.into())
+    }
+}
+
+// ============= NodeName =============
+
+impl NodeName {
```

Comment:
Maybe it should be a generic type with marker, like `Name<Node>`, `Name<Disk>`? So code for them can be unified

Reply:
And then we can make `NodeName = Name<Node>`, `DiskName = Name<Disk>`.

Reply:
Interesting idea. I've implemented it